src/utils/styled_lines_iterator.rs

```diff
@@ -0,0 +1,224 @@
+use utils::LinesIterator;
+use style::*;
+
+use unicode_width::UnicodeWidthStr;
+
+/// Generates rows of styled text in constrained width.
+///
+/// Given styled text and a width constraint, it iterates over
+/// collections of styled spans, each making up a row within the constraint.
+///
+/// # Examples
+///
+/// ```
+/// # use cursive::style::{StyledString, StyledText};
+/// # use cursive::utils::StyledLinesIterator;
+/// # let text_styled: &[StyledString] = &[].as_ref();
+/// let text_plain = text_styled.to_plain();
+/// let spans = StyledLinesIterator::make_spans(text_styled);
+/// // You can now keep `text_plain` and `spans` alive while using `iter`.
```

Comment:
The problem is that we mix direct indices in a string (`StyledSpan`) and `&str` (`StyledString`), and it's not easy to move from one to the other.

Since we can now print content which is not directly present in the input (because of escaping), the indices from `StyledSpan` are no longer a great solution. Keeping a `Cow` in the `StyledSpan` sounds easier.
Actually, I think we could merge `StyledSpan` and `StyledString`. Both refer to a "styled" section from the input text, potentially "rewritten" to account for escaping (hence the `Cow`).

Then, we'd have:

```rust
fn parse(input: &'a str) -> Vec<StyledSpan<'a>>
fn compute_rows(spans: &'b [StyledSpan<'a>]) -> Vec<StyledSpan<'a>>
```

For the `compute_rows`, I'm thinking of updating the existing `LinesIterator` rather than wrapping it. I'm not a big fan of `StyledLinesIterator` taking both the plain text and the spans, and hoping that they'll match. Especially since it doesn't _need_ the plain text.

I'll try to work on updating `LinesIterator` for that this week if I find some time.